fiftyone/core/map/mapper.py

```diff
@@ -168,8 +168,15 @@ def map_samples(
             Iterator[Tuple[bson.ObjectId, R]]: The sample ID and the result of
               the map function for the sample.
         """
+        try:
+            result = check_if_return_is_sample(sample_collection, map_fcn)
```

Comment:
We can do len(sample_collection) to check if a collection is empty, but the call might be expensive for very large datasets.

Reply:
Len for full dataset is fast now. The concern would be with a complex view.
In fact, even getting first() may take a while if the pipeline is complex.
We might want to do something about that, however for this release the fix for empty collection should be fine.
I think a warning but not error seems appropriate, but product may have an opinion @brimoor

Reply:
A validate_output param for example.
Or remove validation and put it lazily on the first sample of the workers' batches

Reply:
> remove validation and put it lazily on the first sample of the workers' batches

This would be okay with me too. We use `warnings.warn()` elsewhere in cases like this to hopefully ensure that a single version of the log message makes it to the user's shell. But it also wouldn't be the worst thing to see `n` versions of this warning because you are definitely not doing the right thing if your `map_fcn` returns `Sample` objects.

Reply:
> I would rather just remove this validation tbh. Yes it is an anti-pattern to return Sample objects. 

This was added because not having the validation still leads to an error, but it is unclear why the error is being raised and happens after work starts happening. Nothing indicates to the user that their `map_fcn` can't return a Sample...the assumption was that the lack of guidance on why your map_samples-empowered workflow wasn't running would have been undesirable. 

However, your call on what makes sense here. We can definitely remove it, especially if this is called out in documentation

Reply:
sgtm ... let's remove the validation for this release? And consider adding it lazily (or some other less invasive check) in develop?

Reply:
I'm onboard with @swheaton's proposal 👍

Reply:
In the interest of time for the release, I have opened this PR: https://github.com/voxel51/fiftyone/pull/5813